src/app/core/cache/object-cache.reducer.ts

```diff
@@ -0,0 +1,59 @@
+import { ObjectCacheAction, ObjectCacheActionTypes, AddToObjectCacheAction, RemoveFromObjectCacheAction } from "./object-cache.actions";
+import { hasValue } from "../../shared/empty.util";
+import { CacheEntry } from "./cache-entry";
+
+export interface CacheableObject {
+  uuid: string;
+}
+
+export class ObjectCacheEntry implements CacheEntry {
+  data: CacheableObject;
+  timeAdded: number;
+  msToLive: number;
+}
+
+export interface ObjectCacheState {
+  [uuid: string]: ObjectCacheEntry
+}
+
+// Object.create(null) ensures the object has no default js properties (e.g. `__proto__`)
+const initialState: ObjectCacheState = Object.create(null);
+
+export const objectCacheReducer = (state = initialState, action: ObjectCacheAction): ObjectCacheState => {
+  switch (action.type) {
+
+    case ObjectCacheActionTypes.ADD: {
+      return addToObjectCache(state, <AddToObjectCacheAction>action);
+    }
+
+    case ObjectCacheActionTypes.REMOVE: {
+      return removeFromObjectCache(state, <RemoveFromObjectCacheAction>action)
+    }
+
+    default: {
+      return state;
+    }
+  }
+};
+
+function addToObjectCache(state: ObjectCacheState, action: AddToObjectCacheAction): ObjectCacheState {
+  return Object.assign({}, state, {
+    [action.payload.objectToCache.uuid]: {
```

Comment:
I had the same issue. The reason is that our mock rest api can't send an error response when it can't find an item. It simply returns a 200 with an empty response. 
But the client should be able to handle unexpected responses as well, so[ I added a check to the data.effects class](https://github.com/DSpace/dspace-angular/pull/62/files#diff-fd2c2e91b0211f4c8e5dd7c329985df5R38) that will throw an error when the returned object is undefined. Eventually this check will be replaced by the [validator](https://github.com/DSpace/dspace-angular/pull/62/files#diff-98b9a305808321337e664dd6c2703c29).